cpp/ql/src/semmle/code/cpp/ir/implementation/raw/internal/IRConstruction.qll

```diff
@@ -61,6 +61,15 @@ private module Cached {
   cached
   predicate hasModeledMemoryResult(Instruction instruction) { none() }
 
+  cached
+  predicate hasConflatedMemoryResult(Instruction instruction) {
+    instruction instanceof UnmodeledDefinitionInstruction
+    or
+    instruction instanceof AliasedDefinitionInstruction
+    or
+    instruction.getOpcode() instanceof Opcode::InitializeNonLocal
```

Comment:
All the uses of `InitializeNonLocal` can hopefully just go away when we remove that opcode. That means this PR has a semantic conflict with @rdmarsh2's work.